_pytest/python.py

```diff
@@ -1152,9 +1153,14 @@ def _idvalset(idx, valset, argnames, idfn, ids):
 def idmaker(argnames, argvalues, idfn=None, ids=None):
     ids = [_idvalset(valindex, valset, argnames, idfn, ids)
            for valindex, valset in enumerate(argvalues)]
-    if len(set(ids)) < len(ids):
-        # user may have provided a bad idfn which means the ids are not unique
-        ids = [str(i) + testid for i, testid in enumerate(ids)]
+    duplicates = [testid for testid in ids if ids.count(testid) > 1]
```

Comment:
At first i used:

   duplicates =  [testid for testid, count in collections.Counter(ids).items() if count > 1]

which is more efficient but Counter is 2.7+.
I will change the code to first check if there are duplicates.